tap_marketo/sync.py

```diff
@@ -431,6 +434,170 @@ def sync_paginated(client, state, stream):
     singer.write_state(state)
     return state, record_count
 
+def sync_leads_describe(client, state, stream):
+    # https://developers.marketo.com/rest-api/endpoint-reference/lead-database-endpoint-reference/#!/Leads/describeUsingGET_2
+    singer.write_schema(stream["tap_stream_id"], stream["schema"], stream["key_properties"])
+    endpoint = "rest/v1/leads/describe.json"
+    data = client.request("GET", endpoint)
+    record_count = 0
+
+    time_extracted = utils.now()
+    for row in data["result"]:
+        record = format_values(stream, row)
+        record_count += 1
+        singer.write_record(stream["tap_stream_id"], record, time_extracted=time_extracted)
+    return state, record_count
+
+def check_if_first_sync(state, tap_stream_id):
+    """
+    checks for initial sync flag stored in the bookmark
+    """
+    first_sync_state = singer.get_bookmark(state, tap_stream_id, INITAL_SYNC_KEY, None)
+    return True if first_sync_state in (None, False) else False
+
+def get_activate_version(state, tap_stream_id):
+    """
+        checks if the sync was interupted in-between an extraction
+        if interupted returns the previous activate version number used
+        else returns a new activate version
+    """
+
+    stream_version = singer.get_bookmark(state, tap_stream_id, ACTIVATE_VERSION_KEY, None)
+    if stream_version is None:
+        stream_version = int(time.time() * 1000)
+        singer.log_info("Using new activate version %s", stream_version)
+    else:
+        singer.log_info("Using interupted activate version %s", stream_version)
+    return stream_version
```

Comment:
```suggestion
    if stream_version:
        singer.log_info("Using interupted activate version %s", stream_version)
    else:
        stream_version = int(time.time() * 1000)
        singer.log_info("Using new activate version %s", stream_version)
    return stream_version
```

Reply:
since this condition is simplified and the changes are tested and validated, changing the sequence of the condition will not have any significant impact.
i would suggest against this change, unless this implementation is causing any adverse impact.